### PR TITLE
Move `admin` from `payments` to `pkg`

### PR DESCRIPTION
**Pull request by mysqto**
Move `admin` pkg from payments and rename to `audit`

### CODE REVIEW

File: audit/mapping.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// RequestToEntity request to entity
+func RequestToEntity(req *Request) (entity Entity) {
+	if req != nil {
+		entity.LastUpdatedBy = req.RequestedBy
+		entity.UpdateReason = req.Reason
+	}
+	return
+}
+
+// EntityToModel base entity to base model
```
Comment (viettd) [inline audit/mapping.go:16]: Shall we explain more abt the `m` parameter?
If I have to use this function, I don't know what I should pass for `m`.
Same for `ActionEntityToAction` below.

File: audit/models.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+type Request struct {
+	// usually an email address, but can be something else
+	// for email see the limit here https://stackoverflow.com/a/574698
+	// for other like names or ids 254 is also enough
+	RequestedBy *string `json:"requestedBy,omitempty"  binding:"required,printascii,max=254"`
+	Reason      *string `json:"reason,omitempty"  binding:"required,printascii,max=512"`
+}
+
+// IChangeRequest general change request interface
+type IChangeRequest interface {
```
Comment (viettd) [inline audit/models.go:17]: `IChangeRequest` is not a good interface name.
How about `IDSetter` if it only has that one function?
Or `ChangeMaker`, `ChangeRequestMaker`?

Comment (mysqto) [reply audit/models.go:17]: @viettd This is the first PR, in the next PR, will have a general function to generic the `bindXXXRequest` like this
``` go
// BindGeneralChangeRequest Bind general change request(Update/Delete)
func BindGeneralChangeRequest(c *gin.Context, ctxKey string, request IChangeRequest) (err error) {
	// try to get from context
	ctxReq, ok := c.Get(ctxKey)
	if ok {
		request, ok = ctxReq.(IChangeRequest)
		if ok {
			return
		}
	}

	// try to bind from request & set to context if ok
	if err = c.ShouldBindJSON(request); err != nil {
		err = errors.New(errors.BadRequest, err)
		return
	}
	var id uint64
	idParam := c.Param("id")
	id, err = strconv.ParseUint(idParam, 10, 64)
	if err != nil || id == 0 {
		err = errors.New(errors.BadRequest, fmt.Sprintf("invalid id [%s]", idParam))
		return
	}
	request.SetID(uint(id))
	c.Set(ctxKey, request)
	return
}
```
I think `IDSetter` or `xxxMaker` maybe not good to understand

Comment (viettd) [reply audit/models.go:17]: I see.I'm ok with this for now since I don't have a better suggestion.

**Review by viettd — APPROVED** [review]

**Pull request merged**